src/crimson/os/seastore/journal.h

```diff
@@ -273,12 +273,42 @@ class Journal {
     record_header_t header,
     bufferlist &bl);
 
+  /// attempts to decode extent infos from bl, return nullopt if unsuccessful
+  std::optional<std::vector<extent_info_t>> try_decode_extent_infos(
+    record_header_t header,
+    bufferlist &bl);
+
+  /**
+   * scan_segment
+   *
+   * Scans bytes_to_read forward from addr to the first record after
+   * addr+bytes_to_read invoking delta_handler and extent_info_handler
+   * on deltas and extent_infos respectively.  deltas, extent_infos
+   * will only be decoded if the corresponding handler is included.
+   */
```

Comment:
could you add a `@return` note in the comment so that readers can understand `SegmentCleaner::do_gc()` better? as i just realized that `P_ADDR_NULL` denotes that scan hits the end of segment.